vcd/resource_vcd_vapp_vm.go

```diff
@@ -834,6 +889,20 @@ func genericResourceVmCreate(d *schema.ResourceData, meta interface{}, vmType ty
 			}
 		}
 
+		err = vm.Refresh()
+		if err != nil {
+			return diag.Errorf("error refreshing VM: %s", err)
+		}
+
+		if enterBiosSetup := d.Get("boot_options.0.enter_bios_setup").(bool); enterBiosSetup {
+			biosSetup := &types.BootOptions{
+				EnterBiosSetup: &enterBiosSetup}
+
+			_, err = vm.UpdateBootOptions(biosSetup)
+			if err != nil {
+				return diag.Errorf("error enabling BIOS setup: %s", err)
+			}
+		}
```

Comment:
We set `EnterBiosSetup` earlier as well, so it enters it here and it would do that on the next power cycle. This if branch is for the case where a user has a powered on VM, changes some options which require the VM to be powered off and then after the power on they still have `EnterBiosSetup` enabled and don't have an inconsistent plan.

Reply:
Now I am thinking of a user workflow and if this is a good approach.
Thinking of this:
* User wants to create a VM and enter BIOS for whatever reason. He changes whatever he wants and lets the VM to boot.

What effecitevely happens is - 
* User creates a VM, enters bios AND the additional part is `this VM is configured so that next VM reboot will get stuck in BIOS`. It looks a bit risky to me - imagine a case where VM is deployed and VM should be rebooted multiple times during guest customization script (which is quite often the case). This will get stuck at bios again until someone manually exits it.



cc: @lvirbalas, @adambarreiro, @dataclouder for thoughts

Reply:
Some remotely similar functionality we handled in such a way https://registry.terraform.io/providers/vmware/vcd/latest/docs/resources/vapp_vm#force

Reply:
Changed now. Changed the argument to `enter_bios_setup_on_next_boot`, it also outputs a warning if `enter_bios_setup_on_next_boot` is set to true.
```
| Warning: TestAccVcdVmAdvancedComputeProperties_empty-19433: After VM Powers on, the enter_bios_setup_on_next_boot flag will be setback to false.
│
│   with vcd_vm.TestAccVcdVmAdvancedComputeProperties_empty-19433,
│   on vcd.TestAccVcdVmAdvancedComputeProperties.tf line 38, in resource "vcd_vm" "TestAccVcdVmAdvancedComputeProperties_empty-19433":
│   38: resource "vcd_vm" "TestAccVcdVmAdvancedComputeProperties_empty-19433" {
│
│ (and one more similar warning elsewhere)
```